src/rules/Rule_15_5_1.cpp

```diff
@@ -0,0 +1,80 @@
+//===-  Rule_15_5_1.cpp - Checker for MISRA C++ 2008 rule 15-5-1-----------===//
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "clang/AST/ASTContext.h"
+#include "clang/AST/OperationKinds.h"
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/Basic/Diagnostic.h"
+#include "misracpp2008.h"
+#include <iostream>
+
+using namespace clang;
+
+template <typename T> Stmt *searchStmt(Stmt *S) {
```

Comment:
Please move this function either in an anonymous namespace or add it as class method to Rule_15_5-1.